### PR TITLE
Add missing functionality in profile page

### PR DESCRIPTION
**Pull request by johngian**

### CODE REVIEW

**Comment by johngian** [comment]
@akatsoulas @comzeradd r?

File: feedthefox/base/static/js/profile.js [hunk 1]
```diff
@@ -7,4 +7,8 @@ jQuery(document).ready(function ($) {
         var term = t.val();
         (term !== '' ? $(target).show() : $(target).hide());
     });
+
+    if ($('#DeviceModal .errorlist').length > 0) {
```
Comment (akatsoulas) [inline feedthefox/base/static/js/profile.js:11]: Why don't you check this inside the template?

File: feedthefox/urls.py [hunk 2]
```diff
@@ -13,6 +13,8 @@
     url(r'^$', 'feedthefox.base.views.home', name='home'),
     url(r'^devices/$', 'feedthefox.devices.views.devices', name='devices'),
     url(r'^device/$', 'feedthefox.devices.views.device', name='device'),
+    url(r'^device_info/(?P<id>\d+)/delete$', 'feedthefox.users.views.delete_device_info',
```
Comment (akatsoulas) [inline feedthefox/urls.py]: s/_/-

File: feedthefox/users/views.py [hunk 3]
```diff
@@ -1,17 +1,43 @@
...
 
 
 @login_required
 def view_profile(request):
     newsletter_form = forms.ReceiveNewsLetterForm(request.POST or None,
-                                                  instance=request.user)
+                                                  instance=request.user,
+                                                  prefix='newsletter')
+    device_info_form = forms.UserDeviceInfoForm(request.POST or None,
+                                                prefix='device_info')
```
Comment (akatsoulas) [inline feedthefox/users/views.py:15]: Why don't you just check the POST request if `device` is present. If not then we have the newsletter form.

**Comment by akatsoulas** [comment]
tested locally and everything seems to be working fine. r+wc :beers:

File: feedthefox/base/static/js/profile.js (hunk 1 above)
Comment (johngian) [inline feedthefox/base/static/js/profile.js:11]: Because in order to show the modal you need to trigger `modal('show')` JS event.

**Pull request merged**